earth_enterprise/src/fusion/autoingest/AssetFactory.h

```diff
@@ -0,0 +1,187 @@
+    using Impl = typename MutableDerivedAssetHandleType::Impl;
+    using AssetStorageType = typename Impl::Base;
+    return MutableDerivedAssetHandleType(std::make_shared<Impl>
+                                            (AssetStorageType::MakeStorage(
+                                        ref_, type_,
+                                        Impl::EXPECTED_SUBTYPE,
+                                        inputs_, meta),
+                                        config));
+  }
+
+  template<class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType Make( const std::string &ref_,
+                                      const std::vector<SharedString>& inputs_,
+                                      const khMetaData &meta,
+                                      const ConfigType &config)
+  {
+    using Impl = typename MutableDerivedAssetHandleType::Impl;
+    using AssetStorageType = typename Impl::Base;
+    return MutableDerivedAssetHandleType(std::make_shared<Impl>
+                                            (AssetStorageType::MakeStorage(
+                                        ref_, Impl::EXPECTED_TYPE,
+                                        Impl::EXPECTED_SUBTYPE,
+                                        inputs_, meta),
+                                        config));
+  }
+
+  template<class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType FindMake(const std::string& ref_,
+                                          AssetDefs::Type type_,
+                                          const khMetaData& meta,
+                                          const ConfigType& config)
+  {
+      MutableDerivedAssetHandleType asset = Find<MutableDerivedAssetHandleType>(ref_, type_);
+      if (asset)
+      {
+          asset->Modify(meta, config);
+          return asset;
+      }
+      return Make<MutableDerivedAssetHandleType, ConfigType>(ref_, type_, meta, config);
+  }
+
+  template<class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType FindMake(const std::string& ref_,
+                                          AssetDefs::Type type_,
+                                          const std::vector<SharedString>& inputs_,
+                                          const khMetaData& meta,
+                                          const ConfigType& config)
+  {
+      MutableDerivedAssetHandleType asset = Find<MutableDerivedAssetHandleType>(ref_, type_);
+      if (asset)
+      {
+          asset->Modify(inputs_, meta, config);
+          return asset;
+      }
+      return Make<MutableDerivedAssetHandleType, ConfigType>(ref_, type_, inputs_, meta, config);
+  }
+
+  template <class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType FindMake(const std::string& ref_,
+                                          const std::vector<SharedString>& inputs_,
+                                          const khMetaData& meta,
```

Comment:
Minor: would be good to have consistency on parameter names - either all with trailing underscores or none.

Reply:
Also, any reason why the next parameter is called `configs` instead of `config`?